### PR TITLE
Move synopsis to case

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
## Description
*This PR moves the cohorts and synopsis from sample to case where it belongs (accord to MM)

### How to prepare for test
- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`
- [x] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [x] do submit a mip-json order with synopsis and cohorts

### Expected test outcome
- [x] check in clinical-api that synopsis and cohorts were saved
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [x] code approved by MM
- [x] tests executed by PG
- [x] "Merge and deploy" approved by PG
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [x] Document in changelog
- [ ] Deploy this branch
- [ ] Inform to MM

### CODE REVIEW

**Comment by patrikgrenfeldt** [comment]
Test result:
image
🆗 samples saved with cohorts and synopsis after submit

File: alembic/versions/fab30255b84f_move_synopsis_to_case.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+Base = declarative_base()
+
+# revision identifiers, used by Alembic.
+revision = "fab30255b84f"
+down_revision = "6d74453565f2"
+branch_labels = None
+depends_on = None
+
+
+class Family(Base):
```
Comment (moonso) [inline alembic/versions/fab30255b84f_move_synopsis_to_case.py:25]: I think that @Mropat and @barrystokman should have a quick look at this as well

File: cg/meta/orders/status.py [hunk 2]
```diff
@@ -145,13 +145,26 @@ def cases_to_status(cls, data: dict) -> dict:
         cases = cls.group_cases(data["samples"])
 
         for case_name, case_samples in cases.items():
-            priority = cls.get_single_value(case_name, case_samples, "priority", "standard")
+            cohorts = set(
+                cohort for sample in case_samples for cohort in sample.get("cohorts", set())
+            )
+            if "" in cohorts:
```
Comment (moonso) [inline cg/meta/orders/status.py:151]: Why this? It seems like `cohorts: Set[Set[str]]` (btw please add type hints), so why are we testing for empty string??

File: cg/meta/orders/status.py [hunk 3]
```diff
@@ -145,13 +145,26 @@ def cases_to_status(cls, data: dict) -> dict:
...
+                cohorts.remove("")
+
+            synopses = set(
```
Comment (moonso) [inline cg/meta/orders/status.py:154]: Type hints

File: cg/meta/orders/status.py [hunk 4]
```diff
@@ -145,13 +145,26 @@ def cases_to_status(cls, data: dict) -> dict:
...
+                synopsis for sample in case_samples for synopsis in sample.get("synopsis", set())
+            )
+            if "" in synopses:
```
Comment (moonso) [inline cg/meta/orders/status.py:157]: Same as above

File: cg/store/api/add.py [hunk 5]
```diff
@@ -149,6 +149,7 @@ def add_case(
         name: str,
         panels: List[str],
         priority: str = "standard",
+        **kwargs,
```
Comment (moonso) [inline cg/store/api/add.py:152]: Are you sure? Should be really hide what we put into the database

**Review by moonso — CHANGES_REQUESTED** [review]
Great, I had some questions

File: cg/meta/orders/status.py (hunk 2 above)
Comment (patrikgrenfeldt) [reply cg/meta/orders/status.py:151]: since it was there and I didn't know how else to get rid of it

File: cg/meta/orders/status.py (hunk 3 above)
Comment (patrikgrenfeldt) [reply cg/meta/orders/status.py:154]: fixed

File: cg/meta/orders/status.py (hunk 4 above)
Comment (patrikgrenfeldt) [reply cg/meta/orders/status.py:157]: yes

File: cg/store/api/add.py (hunk 5 above)
Comment (patrikgrenfeldt) [reply cg/store/api/add.py:156]: maybe not, fixed

File: cg/meta/orders/status.py (hunk 2 above)
Comment (patrikgrenfeldt) [reply cg/meta/orders/status.py]: fixed

File: cg/meta/orders/status.py (hunk 4 above)
Comment (patrikgrenfeldt) [reply cg/meta/orders/status.py]: fixed

**Review by moonso — APPROVED** [review]
Looks good!

**Comment by patrikgrenfeldt** [comment]
@Mropat I get this error from trying to upgrade the cgfixture.db 
```
alembic --config alembic.ini upgrade head
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 6d74453565f2 -> fab30255b84f, move_synopsis_to_case
Traceback (most recent call last):
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
sqlite3.OperationalError: near "DROP": syntax error

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/bin/alembic", line 8, in <module>
    sys.exit(main())
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/config.py", line 581, in main
    CommandLine(prog=prog).main(argv=argv)
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/config.py", line 575, in main
    self.run_cmd(cfg, options)
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/config.py", line 552, in run_cmd
    fn(
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/command.py", line 298, in upgrade
    script.run_env()
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/script/base.py", line 489, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/util/pyfiles.py", line 98, in load_python_file
    module = load_module_py(module_id, path)
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/util/compat.py", line 184, in load_module_py
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "alembic/env.py", line 73, in <module>
    run_migrations_online()
  File "alembic/env.py", line 67, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/runtime/environment.py", line 846, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/runtime/migration.py", line 522, in run_migrations
    step.migration_fn(**kw)
  File "/Users/patrik.grenfeldt/Documents/GitHub/cg/alembic/versions/fab30255b84f_move_synopsis_to_case.py", line 113, in upgrade
    op.drop_column("sample", "_synopsis")
  File "<string>", line 8, in drop_column
  File "<string>", line 3, in drop_column
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/operations/ops.py", line 2049, in drop_column
    return operations.invoke(op)
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/operations/base.py", line 373, in invoke
    return fn(self, operation)
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/operations/toimpl.py", line 80, in drop_column
    operations.impl.drop_column(
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/ddl/impl.py", line 241, in drop_column
    self._exec(base.DropColumn(table_name, column, schema=schema))
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/alembic/ddl/impl.py", line 141, in _exec
    return conn.execute(construct, *multiparams, **params)
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/sql/ddl.py", line 72, in _execute_on_connection
    return connection._execute_ddl(self, multiparams, params)
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1068, in _execute_ddl
    ret = self._execute_context(
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception
    util.raise_(
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
  File "/Users/patrik.grenfeldt/opt/anaconda3/envs/dev38/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 593, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) near "DROP": syntax error
[SQL: ALTER TABLE sample DROP COLUMN _synopsis]
(Background on this error at: http://sqlalche.me/e/13/e3q8)
```
Seems like alembic doesn't translate the drop statement into a dialect that can be executed on the fixture-database, could it be wrong url-specification? 
I'm using `sqlalchemy.url = sqlite:///tests/fixtures/data/cgfixture.db` now

**Comment by patrikgrenfeldt** [comment]
db-upgrade on Hasta:
image

cg upgrade on Hasta:
image

cg upgrade on clinical-db:
image
